### PR TITLE
CDAP 19925 Unit Testing for Grid Page

### PR DESCRIPTION
**Pull request by adityadivami**
# [CDAP-19925]
Unit Test cases for Grid Page

## Description
Added Unit Test cases for Grid Page

## PR Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [x] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [Jira issue #](fill-in.org)
https://cdap.atlassian.net/browse/CDAP-19925

## Test Plan
Not Applicable

## Screenshots
Not Applicable

[CDAP-19925]: https://cdap.atlassian.net/browse/CDAP-19925?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/cdap/components/GridTable/__test__/GridTable.test.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ *  the License.
+ */
+
+import { render } from '@testing-library/react';
+import MyDataPrepApi from 'api/dataprep';
+import React from 'react';
+import { Route, Router, Switch } from 'react-router';
+import rxjs from 'rxjs/operators';
+import history from 'services/history';
+import GridTable from '..';
```
Comment (seanfeld) [inline app/cdap/components/GridTable/__test__/GridTable.test.tsx]: ../index

File: app/cdap/components/GridTable/components/Breadcrumb/__test__/Breadcrumb.test.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+import { Route, Router, Switch } from 'react-router';
+import history from 'services/history';
+import BreadCrumb from '..';
```
Comment (seanfeld) [inline app/cdap/components/GridTable/components/Breadcrumb/__test__/Breadcrumb.test.tsx]: ../index

File: app/cdap/components/GridTable/components/GridHeaderCell/__test__/GridHeaderCell.test.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+import { screen } from '@testing-library/dom';
+import { render } from '@testing-library/react';
+import React from 'react';
+import GridHeaderCell from '..';
```
Comment (seanfeld) [inline app/cdap/components/GridTable/components/GridHeaderCell/__test__/GridHeaderCell.test.tsx]: ../index

File: app/cdap/components/GridTable/components/GridKPICell/__test__/GridKPICell.test.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+import { render } from '@testing-library/react';
+import GridKPICell from '..';
```
Comment (seanfeld) [inline app/cdap/components/GridTable/components/GridKPICell/__test__/GridKPICell.test.tsx]: ../index

File: app/cdap/components/GridTable/components/GridTextCell/__test__/GridTextCell.test.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+import GridTextCell from '..';
```
Comment (seanfeld) [inline app/cdap/components/GridTable/components/GridTextCell/__test__/GridTextCell.test.tsx]: ../index

File: app/cdap/components/GridTable/components/Typography/__test__/TypographyComponent.test.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+import { screen } from '@testing-library/dom';
+import { render } from '@testing-library/react';
+import React from 'react';
+import TypographyComponent from '..';
```
Comment (seanfeld) [inline app/cdap/components/GridTable/components/Typography/__test__/TypographyComponent.test.tsx]: ../index

File: app/cdap/components/GridTable/types.ts [hunk 7]
```diff
@@ -20,7 +20,7 @@ export interface IParams {
 }
 
 export interface IRecords {
-  [key: string]: string;
+  [key: string]: string | Record<string, Record<string, string | number>>;
```
Comment (seanfeld) [inline app/cdap/components/GridTable/types.ts]: this type is changed in a lot of these unit tests prs...?

File: app/cdap/components/GridTable/components/Breadcrumb/__test__/Breadcrumb.test.tsx (hunk 2 above)
Comment (adityadivami) [reply app/cdap/components/GridTable/components/Breadcrumb/__test__/Breadcrumb.test.tsx]: done

File: app/cdap/components/GridTable/components/GridHeaderCell/__test__/GridHeaderCell.test.tsx (hunk 3 above)
Comment (adityadivami) [reply app/cdap/components/GridTable/components/GridHeaderCell/__test__/GridHeaderCell.test.tsx]: done

File: app/cdap/components/GridTable/components/GridKPICell/__test__/GridKPICell.test.tsx (hunk 4 above)
Comment (adityadivami) [reply app/cdap/components/GridTable/components/GridKPICell/__test__/GridKPICell.test.tsx]: done

File: app/cdap/components/GridTable/components/GridTextCell/__test__/GridTextCell.test.tsx (hunk 5 above)
Comment (adityadivami) [reply app/cdap/components/GridTable/components/GridTextCell/__test__/GridTextCell.test.tsx]: done

File: app/cdap/components/GridTable/components/Typography/__test__/TypographyComponent.test.tsx (hunk 6 above)
Comment (adityadivami) [reply app/cdap/components/GridTable/components/Typography/__test__/TypographyComponent.test.tsx]: done

File: app/cdap/components/GridTable/types.ts (hunk 7 above)
Comment (adityadivami) [reply app/cdap/components/GridTable/types.ts]: changing to same in all prs

File: app/cdap/components/GridTable/__test__/GridTable.test.tsx (hunk 1 above)
Comment (seanfeld) [inline app/cdap/components/GridTable/__test__/GridTable.test.tsx]: don't import from '..' -- do `GridTable from  '../index';`

**Review by seanfeld — APPROVED** [review]
Please don't import from '..'

File: app/cdap/components/GridTable/__test__/GridTable.test.tsx (hunk 1 above)
Comment (abhilashDivami) [reply app/cdap/components/GridTable/__test__/GridTable.test.tsx]: Done.

**Review by mrahanjam — APPROVED** [review]

**Pull request merged**